### PR TITLE
Migrated from Algosigner algod to local algodClient

### PR DESCRIPTION
**Pull request by Megha-Dev-19**

### CODE REVIEW

File: src/indexer/index.ts [hunk 1]
```diff
@@ -33,85 +33,53 @@ export const searchForAssets = async (
...
+		let localStateMap = undefined;
+		// parse local state of DAO app for current user to get their details related to app for UI
+		const localState = await indexerClient
+			.lookupAccountAppLocalStates(address)
+			.do();
+		const parsedLocalState = JSON.parse(JSON.stringify(localState));
+
+		if (parsedLocalState && parsedLocalState[LOCAL_STATE]) {
+			const applicationInfo = parsedLocalState[LOCAL_STATE].find(
+				(i: any) => i[APPLICATION_ID] === applicationId
```
Comment (vuvoth) [inline src/indexer/index.ts]:
```suggestion
				(appInfo: any) => appInfo[APPLICATION_ID] === applicationId
```

File: src/indexer/index.ts [hunk 2]
```diff
@@ -122,7 +90,7 @@ export const searchApplicationAndAccount = async () => {
 		return;
 	}
 	const dao_id = daoIdStore.dao_id;
-	const application = await searchForApplication(dao_id).catch((error) => {
+	const application = await getApplicationGlobalState(dao_id).catch((error) => {
```
Comment (vuvoth) [inline src/indexer/index.ts:94]: What is different between `try..catch` and `await...catch`.

File: src/indexer/index.ts [hunk 3]
```diff
@@ -183,45 +163,71 @@ export const isAssetOpted = async (
...
-		if (optedApplicationInfo && optedApplicationInfo["app-local-state"]) {
-			return optedApplicationInfo["app-local-state"].id === application_id;
+		let isApplicationOpted = false;
+		const applicationInfo = await indexerClient
+			.lookupAccountAppLocalStates(address)
+			.do();
+		const parsedApplicationInfo = JSON.parse(JSON.stringify(applicationInfo));
+		if (parsedApplicationInfo && parsedApplicationInfo[LOCAL_STATE]) {
+			const optedApplicationInfo = applicationInfo[LOCAL_STATE].find(
+				(i: any) => (i[APPLICATION_ID] = applicationId)
```
Comment (vuvoth) [inline src/indexer/index.ts]:
```suggestion
				(appInfo: any) => (appInfo[APPLICATION_ID] = applicationId)
```

File: src/indexer/index.ts [hunk 4]
```diff
@@ -183,45 +163,71 @@ export const isAssetOpted = async (
...
+export const getGovASATokenAmount = async (
+	address: string,
+	govtID: number
+): Promise<number> => {
+	try {
+		const assetInfo = await indexerClient.lookupAccountAssets(address).do();
+		let amount = 0;
+		if (assetInfo[ASSETS]) {
+			const govtAssetInfo = assetInfo[ASSETS].find(
+				(i: any) => i[ASSET_ID] === govtID
```
Comment (vuvoth) [inline src/indexer/index.ts]:
```suggestion
				(assetInfo: any) => assetInfo[ASSET_ID] === govtID
```

File: src/indexer/index.ts [hunk 5]
```diff
@@ -166,14 +142,18 @@ export const isAssetOpted = async (
...
-			path: `/v2/accounts/${address}/assets/${asset_id}`,
-		});
-		if (optedAssetInfo && optedAssetInfo["asset-holding"]) {
-			return optedAssetInfo["asset-holding"]["asset-id"] === asset_id;
+		let isAssetOpted = false;
+		const assetInfo = await indexerClient.lookupAccountAssets(address).do();
+
+		if (assetInfo.assets) {
+			const optedAssetInfo = assetInfo.assets.find(
+				(i: any) => i[ASSET_ID] === asset_id
```
Comment (vuvoth) [inline src/indexer/index.ts]:
```suggestion
				(asset: any) => asset[ASSET_ID] === asset_id
```

File: src/indexer/index.ts [hunk 6]
```diff
@@ -166,14 +142,18 @@ export const isAssetOpted = async (
 	asset_id: number
```
Comment (vuvoth) [inline src/indexer/index.ts]: I recommend use camel case style.

File: src/indexer/index.ts [hunk 7]
```diff
@@ -183,45 +163,71 @@ export const isAssetOpted = async (
...
-			throw e;
 		}
+		return amount;
+	} catch (e) {
+		console.log(e);
+		throw e;
 	}
 };
+
+export async function decodeStateMap(
```
Comment (vuvoth) [inline src/indexer/index.ts]: We don't use any `await` in decodeStateMap. Can we remove async keyword?

File: src/indexer/index.ts [hunk 8]
```diff
@@ -183,45 +163,71 @@ export const isAssetOpted = async (
...
+	state: Array<any>
+): Promise<Map<Key, StateValue>> {
+	const stateMap = new Map<Key, StateValue>();
+	if (state) {
+		for (const g of state) {
+			const key = Buffer.from(g.key, "base64").toString();
+			if (g.value.type === 1) {
```
Comment (vuvoth) [inline src/indexer/index.ts]: What is `1` meaning in this case?

**Review by vuvoth — COMMENTED** [review]
Nice work! Left few comments.

File: src/indexer/index.ts (hunk 2 above)
Comment (Megha-Dev-19) [reply src/indexer/index.ts:94]: actually there are two ways to handle asynchronous promise either by using async, await/catch or by then/catch 
async, await was introduced later in ES7.
you can look at it https://www.smashingmagazine.com/2020/11/comparison-async-await-versus-then-catch/

File: src/indexer/index.ts (hunk 6 above)
Comment (Megha-Dev-19) [reply src/indexer/index.ts]: Sure, will update

File: src/indexer/index.ts (hunk 7 above)
Comment (Megha-Dev-19) [reply src/indexer/index.ts]: yes, will update

File: src/indexer/index.ts (hunk 8 above)
Comment (Megha-Dev-19) [reply src/indexer/index.ts]: it signifies that it is bytes type not uint

**Review by vuvoth — APPROVED** [review]

**Pull request merged**